Added HTML → Markdown conversion tool

AezanPathan opened the pull request:
## Purpose
<!-- Describe the intention of the changes being proposed. What problem does it solve or functionality does it add? -->
* Adds a new tool class HtmlToMarkdownTool to convert HTML back to Markdown.
This allows users to perform two-way conversion between Markdown and HTML using separate tools (MarkdownToHtmlTool and HtmlToMarkdownTool).

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[x] New feature to existing sample
[ ] New sample
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## README updated?

The top-level readme for this repo contains a link to each sample in the repo. If you're adding a new sample did you update the readme?
<!-- Mark one with an "x". -->
```
[ ] Yes
[x] No
[ ] N/A
```

## How to Test
*  Get the code

```
git clone https://github.com/AezanPathan/mcp-dotnet-samples.git
cd mcp-dotnet-samples
git checkout feature/reverse-markdown

```

* Test the code
<!-- Add steps to run the tests suite and/or manually test -->
```
Use the HtmlToMarkdownTool.ConvertAsync method to convert an HTML string back to Markdown.

Verify that HTML input is correctly converted to Markdown output.

Existing ConvertAsync method for Markdown → HTML should continue to work without changes.
```

## What to Check
Verify that the following are valid
* HTML content is properly converted back to Markdown using HtmlToMarkdownTool.

* Existing Markdown → HTML conversion still works.

* No errors or exceptions occur when using either conversion method.

## Other Information
<!-- Add any other helpful information that may be needed here. -->
* Uses the ReverseMarkdown NuGet package for HTML → Markdown conversion.

* Tested locally and works alongside the existing Markdown → HTML pipeline.

AezanPathan commented:
@microsoft-github-policy-service agree

justinyoo reviewed (changes requested):
@AezanPathan Thanks for this PR! However, I'd like you to add another tool class like `HtmlToMarkdownTool.cs` and put your logic there. Would you please take a look please?

AezanPathan commented:
Hi @justinyoo, I’ve updated the PR — moved the logic into the new HtmlToMarkdownTool class as suggested and adjusted the PR description accordingly. Please review when you get a chance. Thanks for the guidance!
